presto-main-base/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -141,10 +143,23 @@ else if (context.getTimeZoneId() != null) {
         return sessionBuilder;
     }
 
-    private Identity authenticateIdentity(QueryId queryId, SessionContext context)
+    private Identity authenticateIdentity(QueryId queryId, SessionContext context, String query)
     {
+        try {
+            validateQueryToken(accessControl, queryId, context, query);
+
+            if (isSkipPermissionsCheckEnabled(accessControl)) {
```

Comment:
what is the purpose of skipPermissionsCheckEnabled?  I think this is because we only want to skip if validateQueryToken was successful.  However, most of the permissions get checked elsewhere (e.g. checking column access), so not sure how useful this is.